src/test/run-pass/generator/auxiliary/xcrate-reachable.rs

```diff
@@ -8,15 +8,16 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-#![feature(generators, generator_trait)]
+#![feature(generators, generator_trait, pin)]
 
+use std::marker::Unpin;
 use std::ops::Generator;
 
 fn msg() -> u32 {
     0
 }
 
-pub fn foo() -> impl Generator<Yield=(), Return=u32> {
+pub fn foo() -> impl Generator<Yield=(), Return=u32> + Unpin {
```

Comment:
Why did you add the `Unpin` bound here and on other tests?

Reply:
`Unpin` is necessary to use `Pin::new`, otherwise the tests would need to use the unsafe `Pin::new_unchecked` (this may not actually be necessary with `impl Trait`, `Unpin` probably leaks like other auto-traits, I didn't check that).

Reply:
Yeah, I was expecting `Unpin` to leak here, so this change shouldn't be necessary.

Reply:
Removed the leaking `Unpin`s.